src/pyhf/workspace.py

```diff
@@ -743,3 +743,42 @@ def sorted(cls, workspace):
         newspec['observations'].sort(key=lambda e: e['name'])
 
         return cls(newspec)
+
+    @classmethod
+    def build(cls, model, data, name='measurement'):
+        """
+        Build a workspace from model and data.
+
+        Args:
+            model (~pyhf.pdf.Model): A model to store into a workspace
+            data (tensor): A array holding observations to store into a workspace
+            name (str): The name of the workspace measurement
+
+        Returns:
+            ~pyhf.workspace.Workspace: A new workspace object
+
+        """
+        workspace = copy.deepcopy(model.spec)
+        workspace['version'] = '1.0.0'
+        workspace['measurements'] = [
+            {
+                'name': name,
+                'config': {
+                    'poi': model.config.poi_name,
+                    'parameters': [
+                        {
+                            "bounds": [list(x) for x in v['paramset'].suggested_bounds],
+                            "inits": v['paramset'].suggested_init,
+                            "fixed": v['paramset'].fixed,
+                            "name": k,
+                        }
```

Comment:
this works, but i'm not sure whether we can avoid being very verbose here. In standard workspaces from xml2json we only add stuff here that is "non-default' while this will dump all parameter configurations even the default ones

Reply:
@kratsg

Reply:
alternatively one could use `model.spec['parameters']` but a) i don't know why parameters is part of the model spec in the first place and b) we strip out the names of the parameter sets.

Reply:
Can we compare against the default and only add in non-default settings? This should be possible based on the code you have

```
parameters = []
for par_name, par_config in model.config.par_map.items():
  _par_config = {}
  bounds = par_config['paramset'].suggested_bounds
  inits = par_config['paramset'].suggested_init
  fixed = par_config['paramset'].fixed
  default_parset = getattr(pyhf.modifiers, dict(model.config.modifiers)[par_name]).required_parset([0]*len(inits), [0]*len(inits))
  if bounds != list(default_parset['bounds']):
    _par_config['bounds'] = bounds
  if inits != list(default_parset['inits']):
    _par_config['inits'] = inits
  if fixed != _par_config['fixed']:
    _par_config['fixed'] = fixed
  parameters.append(_par_config)
```